### ISSUE TITLE
About log softmax

### ISSUE THREAD

**Issue by WilliamTambellini** [intel/mkl-dnn#512]
From what I ve read 
https://intel.github.io/mkl-dnn/dev_guide_softmax.html
and some sources
https://github.com/intel/mkl-dnn/blob/master/src/cpu/ref_softmax.cpp

it looks like mkldnn is computing the 'regular' softmax.
For some NN, it is useful/needed to use the log softmax.
Sadly, just applying log to the softmax output is known to be numerically unstable.
Usual implementation apply log 'internally' : 
Refs:
TF:
https://github.com/petewarden/tensorflow_makefile/blob/49c08e4d4ff3b6e7d99374dc2fbf8b358150ef9c/tensorflow/core/kernels/softmax_op_functor.h#L44
PT:
https://github.com/pytorch/pytorch/blob/6ffacd5f02d78a7b3fc5ce473879b36900d2ecb4/torch/nn/functional.py#L1292
Cudnn:
https://docs.nvidia.com/deeplearning/sdk/cudnn-developer-guide/index.html#cudnnSoftmaxAlgorithm_t

Could anyone first confirm that today logsoftmax not available in mkldnn ?
Kind

-----------------------------------------------------------------------------

### Environment
* version :  1.0

### Steps to reproduce
https://gist.github.com/WilliamTambellini/907b8ab7ac85d2f0914350c009fc7213

### Actual behavior
regular softmax

### Expected behavior
logsoftmax after adding a "softmax kind" option to the softmax descriptior.

**Comment by emfomenk** [comment]
Hi @WilliamTambellini,

Yeah, I confirm we don't have `LogSoftmax`.
This might be a good candidate for adding to the library.

However, I want to note that Intel MKL-DNN implements the *stable* algorithm of Softmax (the one that subtracts the maximum of values). Hence applying the `log` to the `Softmax` should not give completely meaningless result. At least that might be a quick work-around of the missing feature.

**Comment by eladdidy** [comment]
Hi, is there eltwize algorithm for simple log, or I need to use some other math code?
Thanks

**Comment by emfomenk** [comment]
Hi @eladdidy,

No, we don't have regular `log`.
The list of the available eltwises can be found [here](https://intel.github.io/mkl-dnn/dev_guide_eltwise.html).

**Comment by dzarukin** [comment]
Hi @WilliamTambellini and @eladdidy,

LogSoftmax support out-of-the-box will be available soon via separate API, hopefully in v1.2.

**Comment by vpirogov** [comment]
The feature is implemented in 2b02c748364b09f0ef2e7ebfd8f2e0d8e774fe1c. Documentation is available in the [developer guide](http://intel.github.io/mkl-dnn/dev_guide_logsoftmax.html).
